### PR TITLE
fix(dashmate): zero ssl verification passes without being verified

### PR DESCRIPTION
**Pull request by shumkov**
## Issue being fixed or feature implemented

ZeroSSL certificate verification passes without actually verifying the certificate.

## What was done?

- Make sure that verification server is responding before we request verification from ZeroSSL
- Remove wrong automate retries on verification
- Display error from undocumented error response from ZeroSSL

## How Has This Been Tested?

Obtain certificates on with open and close 80 port

## Breaking Changes

None

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have added "!" to the title and described breaking changes in the corresponding section if my code contains any
- [x] I have made corresponding changes to the documentation if needed

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

## Summary by CodeRabbit

- **New Features**
	- Added a server responsiveness check during the verification process for enhanced reliability.
	- Introduced a new method to wait for server response, improving validation workflow.

- **Bug Fixes**
	- Enhanced error messages for troubleshooting during certificate verification, providing clearer guidance for users.

- **Documentation**
	- Updated error handling to include more contextual information, aiding user understanding during issues.

### CODE REVIEW

**Review by lklimek — APPROVED** [review]

File: packages/dashmate/src/listr/tasks/ssl/VerificationServer.js [hunk 1]
```diff
@@ -160,6 +168,31 @@ export default class VerificationServer {
     this.container = null;
   }
 
+  async waitForServerIsResponding() {
+    const MAX_WAIT_TIME = 10000; // Maximum wait time in milliseconds
+    const INTERVAL = 500; // Interval to check in milliseconds
+    const FETCH_TIMEOUT = 2000; // Timeout for each fetch in ms
```
[the remark this answers is not in the document]
Comment (shumkov) [reply packages/dashmate/src/listr/tasks/ssl/VerificationServer.js:174]: It's a local server.

**Comment by shumkov** [comment]
@coderabbit review

File: packages/dashmate/src/listr/tasks/ssl/zerossl/obtainZeroSSLCertificateTaskFactory.js [hunk 2]
```diff
@@ -189,38 +200,37 @@ export default function obtainZeroSSLCertificateTaskFactory(
...
-                    enabled: 'Yes',
-                    disabled: 'No',
-                    initial: true,
-                  });
+                // Get the error message from details if it exists
+                if (e.type === 'domain_control_validation_failed' && e.details[ctx.externalIp]) {
+                  const errorDetails = Object.values(e.details[ctx.externalIp])[0];
+                  if (errorDetails.error) {
+                    errorMessage = errorDetails.error_info;
+                  }
```
[the remark this answers is not in the document]
Comment (shumkov) [reply packages/dashmate/src/listr/tasks/ssl/zerossl/obtainZeroSSLCertificateTaskFactory.js:213]:
```
Object.values(e.details[ctx.externalIp]).length > 0
```
This condition is good, but If we have error detail, I believe the structure will be the same.

File: packages/dashmate/src/listr/tasks/ssl/VerificationServer.js [hunk 3]
```diff
@@ -3,8 +3,14 @@ import path from 'path';
 import dots from 'dot';
 import os from 'os';
 import { TEMPLATES_DIR } from '../../../constants.js';
+import wait from '../../../util/wait.js';
```
[the remark this answers is not in the document]
Comment (shumkov) [reply packages/dashmate/src/listr/tasks/ssl/VerificationServer.js:6]: We have minimum requirements Node.JS 20  for the project, I don't think we need to support previous versions in this case.

**Pull request merged**